### ISSUE TITLE
Cannot build with latest nightly rustc

### ISSUE THREAD

**Issue by a0x** [murarth/rusti#74]
I installed the latest rustc nightly, which is `rustc 1.10.0-nightly (645dd013a 2016-04-24)`, and turned out that the project cannot be built.

```
   Compiling rusti v0.0.1 (file:///home/a0x  /workspace/rust/nightly/rusti)
src/rusti/exec.rs:335:27: 335:43 error: unresolved name `driver::make_map` [E0425]
src/rusti/exec.rs:335             let ast_map = driver::make_map(&sess, &mut forest);
                                                ^~~~~~~~~~~~~~~~
src/rusti/exec.rs:335:27: 335:43 help: run `rustc --explain E0425` to see a detailed explanation
src/rusti/exec.rs:401:27: 401:43 error: unresolved name `driver::make_map` [E0425]
src/rusti/exec.rs:401             let ast_map = driver::make_map(&sess, &mut forest);
                                                ^~~~~~~~~~~~~~~~
src/rusti/exec.rs:401:27: 401:43 help: run `rustc --explain E0425` to see a detailed explanation
src/rusti/exec.rs:331:23: 331:64 error: this function takes 3 parameters but 2 parameters were supplied [E0061]
src/rusti/exec.rs:331             let lcx = LoweringContext::new(&sess, Some(&krate));
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/rusti/exec.rs:331:23: 331:64 help: run `rustc --explain E0061` to see a detailed explanation
src/rusti/exec.rs:337:13: 361:19 error: this function takes 6 parameters but 7 parameters were supplied [E0061]
src/rusti/exec.rs:337             driver::phase_3_run_analysis_passes(
src/rusti/exec.rs:338                 &sess, &cstore, ast_map, &arenas, id, MakeGlobMap::No,
src/rusti/exec.rs:339                 |tcx, mir_map, analysis, _| {
src/rusti/exec.rs:340                     tcx.sess.abort_if_errors();
src/rusti/exec.rs:341 
src/rusti/exec.rs:342                     let trans = driver::phase_4_translate_to_llvm(
                      ...
src/rusti/exec.rs:337:13: 361:19 help: run `rustc --explain E0061` to see a detailed explanation
src/rusti/exec.rs:340:21: 340:29 error: the type of this value must be known in this context
src/rusti/exec.rs:340                     tcx.sess.abort_if_errors();
                                          ^~~~~~~~
src/rusti/exec.rs:343:30: 343:63 error: the type of this value must be known in this context
src/rusti/exec.rs:343                         tcx, mir_map.expect("mir_map is None"), analysis);
                                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/rusti/exec.rs:351:32: 351:50 error: the type of this value must be known in this context
src/rusti/exec.rs:351                     let deps = crates.into_iter().rev()
                                                     ^~~~~~~~~~~~~~~~~~
src/rusti/exec.rs:365:43: 365:66 error: the type of this value must be known in this context
src/rusti/exec.rs:365     r.and_then(|r| r).map(|(modp, deps)| (modp as llvm::ModuleRef, deps))
                                                                ^~~~~~~~~~~~~~~~~~~~~~~
src/rusti/exec.rs:397:23: 397:64 error: this function takes 3 parameters but 2 parameters were supplied [E0061]
src/rusti/exec.rs:397             let lcx = LoweringContext::new(&sess, Some(&krate));
                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/rusti/exec.rs:397:23: 397:64 help: run `rustc --explain E0061` to see a detailed explanation
src/rusti/exec.rs:403:13: 408:23 error: this function takes 6 parameters but 7 parameters were supplied [E0061]
src/rusti/exec.rs:403             driver::phase_3_run_analysis_passes(
src/rusti/exec.rs:404                 &sess, &cstore, ast_map, &arenas, id, MakeGlobMap::No,
src/rusti/exec.rs:405                     |tcx, _mir_map, analysis, _| {
src/rusti/exec.rs:406                         let _ignore = tcx.dep_graph.in_ignore();
src/rusti/exec.rs:407                         f(&krate, tcx, analysis)
src/rusti/exec.rs:408                     })
src/rusti/exec.rs:403:13: 408:23 help: run `rustc --explain E0061` to see a detailed explanation
src/rusti/exec.rs:406:39: 406:52 error: the type of this value must be known in this context
src/rusti/exec.rs:406                         let _ignore = tcx.dep_graph.in_ignore();
                                                            ^~~~~~~~~~~~~
error: aborting due to 9 previous errors
error: Could not compile `rusti`.

```
Among those errors, I found out `rustc_driver` module is no longer exist in the latest rustc. I think this is a truly serious problem.

**Comment by murarth** [comment]
Internal `rustc` APIs are continuously subject to change. Often, this takes the form of crates or modules being moved or renamed. Rarely is any necessary functionality lost. This will be fixed shortly. Thank you for filing the issue.

**Comment by murarth** [comment]
This is now fixed in the latest commit: b1873f8c8adbad02a59ccb8993ed1dd2bc480a68
